### PR TITLE
[tune] debug py37 build

### PR DESCRIPTION
**Pull request by richardliaw**
## Why are these changes needed?

* Lint is currently failing

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ci/travis/install-dependencies.sh [hunk 1]
```diff
@@ -237,10 +237,11 @@ install_dependencies() {
   install_base
   install_toolchains
   install_nvm
-  install_pip
+  install_upgrade_pip
 
   if [ -n "${PYTHON-}" ] || [ "${LINT-}" = 1 ]; then
     install_miniconda
+    install_upgrade_pip
```
Comment (richardliaw) [inline ci/travis/install-dependencies.sh:244]:
```suggestion
    install_miniconda
    # Upgrade the miniconda pip.
    install_upgrade_pip
```

**Comment by amogkam** [comment]
Ah nice catch! Do you know why the Python 3.6 tests don't go through the miniconda codepath? Or I guess what is `if [ -n "${PYTHON-}" ]` checking for?

**Comment by richardliaw** [comment]
They do go through the miniconda path. On 3.7, i think we install miniconda but don't set pip to the right miniconda pip.

On 3.6 i think we install miniconda, and we use the miniconda pip going forward. 

In any case, this actually passes tests: https://travis-ci.com/github/ray-project/ray/builds/206209467

**Review by krfricke — APPROVED** [review]

**Review by amogkam — APPROVED** [review]

**Comment by amogkam** [comment]
@richardliaw This is great! I'm just asking so I can understand where the 3.6 and 3.7 tests diverge. Do you know where/why 

> On 3.7, i think we install miniconda but don't set pip to the right miniconda pip

is happening? I had spent some time trying to figure this out and from our install-dependencies script it did not seem like there should've been any pip difference between the 3.6 and 3.7 tests.

**Comment by richardliaw** [comment]
My hypothesis is that the 3.6 miniconda env has the newest version of pip but not on 3.7. 

I was able to verify that after miniconda installation, the `pip --version` on our 3.7 was not the latest pip.

However, it is actually really hard to give you a full debug because every tweak to the script requires a CI build, which can queue for multiple hours...

**Pull request merged**